src/pages/Hacker/Pass.tsx

```diff
@@ -57,69 +49,82 @@ const HackPassWrapper = styled.div`
     }
   }
 
-  button[type="submit"] {
+  button[type='submit'] {
     position: relative;
     top: calc(-50px - 38px / 2);
   }
 `;
 
-class HackPassContainer extends React.Component<{}, IDashboardState> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      account: null,
-      hacker: null,
-      qrData: '',
-      loadingHacker: true,
-      downloadingPass: false,
-    };
-    this.handleDownloadPass = this.handleDownloadPass.bind(this);
-  }
-  public async componentDidMount() {
-    try {
-      const account = (await Account.getSelf()).data.data;
-      const hacker = (await Hacker.getSelf()).data.data;
-      const qrData = await generateHackerQRCode(hacker);
-      this.setState({ account, hacker, qrData });
-    } catch (e) {
-      if (e && e.data) {
-        ValidationErrorGenerator(e.data);
+const HackPassPage: React.FC = () => {
+  // Store the user's data
+  const [account, setAccount] = useState<IAccount | null>(null);
+  const [hacker, setHacker] = useState<IHacker | null>(null);
+  const [qrData, setQrData] = useState<string>('');
+
+  // Is the hacker data currently being fetched from server?
+  const [loadingHacker, setLoadingHacker] = useState<boolean>(true);
+
+  // Is the pass currently being downloaded
+  const [downloadingPass, setDownloadingPass] = useState<boolean>(true);
+
+  // When the component mounts, fetch data associate with this hacker from server
+  useEffect(() => {
+    (async () => {
+      try {
+        const account = (await Account.getSelf()).data.data;
+        const hacker = (await Hacker.getSelf()).data.data;
+        const qrData = await generateHackerQRCode(hacker);
+        setAccount(account);
+        setHacker(hacker);
+        setQrData(qrData);
+      } catch (e) {
+        if (e && e.data) {
+          ValidationErrorGenerator(e.data);
+        }
+      } finally {
+        setLoadingHacker(false);
       }
-    } finally {
-      this.setState({ loadingHacker: false });
-    }
-  }
-  public render() {
-    const { qrData, account, hacker, downloadingPass } = this.state;
-    if (qrData && account && hacker) {
-      return (
-        <HackPassWrapper>
-          <h1>Your HackPass</h1>
-          <Pass account={account} hacker={hacker} qrData={qrData} />
-          <SubmitBtn
-            onClick={this.handleDownloadPass}
-            isLoading={downloadingPass}
-          >
-            Download pass
-          </SubmitBtn>
-        </HackPassWrapper>
-      );
-    }
-    if (this.state.loadingHacker) {
-      return <h1>Loading...</h1>;
-    }
-    return <h1>Error</h1>;
-  }
+    })();
+  }, []);
 
-  private async handleDownloadPass(): Promise<void> {
-    const { account, hacker } = this.state;
+  // Generage hackpass pdf and download it to the user's computer
+  const handleDownloadPass = async () => {
```

Comment:
private?

Reply:
This one lost it's return type of `: Promise<void>` too